src/devices/somfy.c

```diff
@@ -0,0 +1,178 @@
+    char address[7];
+    uint8_t control = 0;
+    uint8_t random = 0;
+
+    for (int i = 0; i < bitbuffer->num_rows; i++) {
+        if (bitbuffer->bits_per_row[i] > 170) {
+            is_retransmission = 1;
+            decode_row = i;
+            data_start = 65;
+            preamble_pattern = (const uint8_t *) "\xf0\xf0\xf0\xf0\xf0\xf0\xf0\xff";
+            preamble_pattern_bit_length = 64;
+            break;
+        } else if (bitbuffer->bits_per_row[i] > 130) {
+            is_retransmission = 0;
+            decode_row = i;
+            data_start = 25;
+            preamble_pattern = (const uint8_t *) "\xf0\xf0\xff";
+            preamble_pattern_bit_length = 24;
+            break;
+        }
+    }
+
+    if (data_start == 0)
+        return DECODE_FAIL_SANITY;
+
+    if (bitbuffer_search(bitbuffer, decode_row, 0, preamble_pattern, preamble_pattern_bit_length) != 0)
+        return DECODE_FAIL_SANITY;
+
+    if (bitbuffer_manchester_decode(bitbuffer, decode_row, data_start, &bitbuffer_decoded, 56) - data_start < 56)
+        if (bitbuffer_manchester_decode(bitbuffer, decode_row, data_start - 1, &bitbuffer_decoded, 56) - data_start + 1 < 56)
+            return DECODE_FAIL_MIC;
+
+    bitbuffer_extract_bytes(&bitbuffer_decoded, 0, 0, message_bytes, 56);
+
+    // descramble
+    for (int i = 6; i >= 0; i--)
+        message_bytes[i] = message_bytes[i] ^ ((i == 0) ? 0 : message_bytes[i-1]);
+
+    // calculate checksum
+    for (int i = 0; i < 7; i++)
+        chksum ^= (message_bytes[i] & 0xf) ^ (message_bytes[i] >> 4);
+
+    // extract random
+    random = message_bytes[0];
+
+    // extract control
+    control = (message_bytes[1] & 0xf0) >> 4;
+
+    // extract counter independent of system's byte order
+    counter = message_bytes[3] + message_bytes[2] * 255;
+
+    // extract address bytes into hexstring
+    snprintf(address, sizeof(address), "%02x%02x%02x", message_bytes[4], message_bytes[5], message_bytes[6]);
+
+    data = data_make(
+            "model",          "",               DATA_STRING, _X("Somfy-RTS","Somfy RTS"),
+            "random",         "Random",         DATA_INT,    random,
+            "control",        "Control",        DATA_STRING, control_str[control],
+            "checksum",       "Checksum",       DATA_STRING, chksum == 0 ? "OK" : "NOK",
+            "counter",        "Counter",        DATA_INT,    counter,
+            "address",        "Address",        DATA_STRING, address,
+            "retransmission", "Retransmission", DATA_STRING, is_retransmission ? "TRUE" : "FALSE",
+            NULL);
+    decoder_output_data(decoder, data);
+
+    return 1;
+}
+
+static char *output_fields[] = {
+    "model",
+    "random",
+    "control",
+    "checksum",
+    "counter",
+    "address",
+    "retransmission",
+    NULL
+};
+
+// rtl_433 -r g001_433.414M_250k.cu8 -X "n=somfy-test,m=OOK_PCM,s=604,l=604,t=40,r=10000,g=3000,y=2416"
+
+r_device somfy_rts = {
+    .name           = "Somfy RTS",
```

Comment:
indent twice (8 spaces)
Perhaps "Somy RTS remote control"?

Reply:
I would stay with "Somfy RTS" as there are other products besides remote controls that also use that protocol (https://asset.somfy.com/Document/4e5d6a89-a67c-4c7e-b303-3cd59a982e9a_RTS_Pocket_Guide_6-2019.pdf). For example, sun and wind sensors. I just do not have them to generate sample traces.